sample-3.md

```diff
@@ -0,0 +1,536 @@
+      labels: *labels
+    spec:
+      containers:
+        - name: event-display
+          # Source code: https://github.com/knative/eventing-sources/blob/release-0.6/cmd/event_display/main.go
+          image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display@sha256:37ace92b63fc516ad4c8331b6b3b2d84e4ab2d8ba898e387c0b6f68f0e3081c4
+
+---
+
+# Service pointing at the previous Deployment. This will be the target for event
+# consumption.
+kind: Service
+apiVersion: v1
+metadata:
+  name: event-display-test
+spec:
+  selector:
+    app: event-display-test
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8080
+END
+```
+
+Let's wait for the `Deployment`s to be ready, as shown by their Available status
+condition.
+
+```shell
+kubectl -n $K_NAMESPACE get deployments event-display-dev event-display-test                                                                                                                                            +54 10:30 ❰─┘
+```
+
+This should return something like:
+```shell
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+event-display-dev    1         1         1            1           16m
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+event-display-test   1         1         1            1           16m
+```
+
+In particular, we want the number in the `DESIRED` column to match the number in the
+`AVAILABLE` column. This may take a few minutes. If after three minutes, it the
+numbers still do not match, then investigate why. 
+
+TODO Add an investigation guide.
+
+## Triggers
+
+Earlier we said that different event consumers can register their intereset with a
+`Broker`. That interest is registered by `Trigger` objects. A `Trigger` object
+logically breaks into two pieces:
+- Filter - What events I am interested in.
+- Subscriber - Where should those events be sent.
+
+### Filter
+
+The Filter can be seen in `spec.filter`. For example:
+```yaml
+spec:
+  filter:
+    sourceAndType:
+      type: knative.eventing.hello_world
+      source: https://knative.eventing.dev
+```
+
+Every valid CloudEvent has attributes named `Type` and `Source`. `Trigger`s allow
+you to specify interest in specific CloudEvents by matching the CloudEvents `Type`
+and `Source`.  
+
+For example, if we want all CloudEvents of type `foo`, regardless of their `Source`,
+then we could use the following filter.
+
+```yaml
+spec:
+  filter:
+    sourceAndType:
+      type: foo
+```
+
+Or if we want all CloudEvents with source `bar`, regardless of their `Type`, then
+we could use the following filter:
+
+```yaml
+spec:
+  filter:
+    sourceAndType:
+      source: bar
+```
+
+### Subscriber
+
+Subscriber describes where to send CloudEvents. Once the filter has matched a
+CloudEvent, then the Subscriber describes where it is sent. Subscriber is an
+Object Reference to an `Addressable`. Normally it is a `Broker`, `Channel`, 
+K8s `Service`, or Knative `Service`.
```

Comment:
Link to respective pages.

Reply:
Isn't a Knative service Addressable? Just asking for my curiosity, for a user I think we should mention it regardless.

Reply:
Yes, a Knative Service is addressable.

Reply:
Done.